pkg/agent/multicast/mcast_socket.go

```diff
@@ -0,0 +1,127 @@
+//go:build linux
```

Comment:
if `multicastsyscall` related functions is moved to mcast_socket_unix, do we still need this build tag?

Reply:
Updated, I renamed `mcast_socket` to `mcast_socket_linux`since most code in mcast_socket are platform-specific.